src/python/pants/backend/codegen/protobuf/java/rules_integration_test.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.codegen.protobuf.java.rules import GenerateJavaFromProtobufRequest
+from pants.backend.codegen.protobuf.java.rules import rules as protobuf_rules
+from pants.backend.codegen.protobuf.python.python_protobuf_subsystem import (
+    rules as protobuf_subsystem_rules,
+)
+from pants.backend.codegen.protobuf.target_types import (
+    ProtobufSourceField,
+    ProtobufSourcesGeneratorTarget,
+)
+from pants.backend.codegen.protobuf.target_types import rules as target_types_rules
+from pants.core.util_rules import stripped_source_files
+from pants.engine.addresses import Address
+from pants.engine.target import GeneratedSources, HydratedSources, HydrateSourcesRequest
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+GRPC_PROTO_STANZA = """
+syntax = "proto3";
+
+package dir1;
+
+// The greeter service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+"""
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *protobuf_rules(),
+            *protobuf_subsystem_rules(),
+            *stripped_source_files.rules(),
+            *target_types_rules(),
+            QueryRule(HydratedSources, [HydrateSourcesRequest]),
+            QueryRule(GeneratedSources, [GenerateJavaFromProtobufRequest]),
+        ],
+        target_types=[ProtobufSourcesGeneratorTarget],
+    )
+
+
+def assert_files_generated(
+    rule_runner: RuleRunner,
+    address: Address,
+    *,
+    expected_files: list[str],
+    source_roots: list[str],
+    mypy: bool = False,
```

Comment:
Bad copy pasta

Reply:
Will fix in separate PR.

Reply:
https://github.com/pantsbuild/pants/pull/14006